client/state/themes/test/reducer.js

```diff
@@ -0,0 +1,405 @@
+import {
+	THEME_REQUEST,
+	THEME_REQUEST_SUCCESS,
+	THEME_REQUEST_FAILURE,
+	THEMES_REQUEST,
+	THEMES_REQUEST_FAILURE,
+	THEMES_REQUEST_SUCCESS,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+import reducer, {
+	queryRequests,
+	queries,
+	themeRequests
+} from '../reducer';
+import ThemeQueryManager from 'lib/query-manager/theme';
+
+const twentysixteen = {
+	id: 'twentysixteen',
+	name: 'Twenty Sixteen',
+	author: 'the WordPress team',
+	screenshot: 'https://i0.wp.com/theme.wordpress.com/wp-content/themes/pub/twentysixteen/screenshot.png',
+	stylesheet: 'pub/twentysixteen',
+	demo_uri: 'https://twentysixteendemo.wordpress.com/',
+	author_uri: 'https://wordpress.org/'
+};
+
+const mood = {
+	id: 'mood',
+	name: 'Mood',
+	author: 'Automattic',
+	screenshot: 'mood.jpg',
+	price: '$20',
+	stylesheet: 'premium/mood',
+	demo_uri: 'https://mooddemo.wordpress.com/',
+	author_uri: 'https://wordpress.com/themes/'
+};
+
+describe( 'reducer', () => {
+	useSandbox( ( sandbox ) => {
+		sandbox.stub( console, 'warn' );
+	} );
+
+	it( 'should include expected keys in return value', () => {
+		expect( reducer( undefined, {} ) ).to.have.keys( [
+			// Old reducers
+			'themes',
+			'themeDetails',
+			'themesList',
+			// New reducers
+			//'queries',
+			//'queryRequests',
+			//'themeRequests',
+			'currentTheme',
+			'themesUI'
+		] );
+	} );
+
+	describe( '#queryRequests()', () => {
+		it( 'should default to an empty object', () => {
+			const state = queryRequests( undefined, {} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		// TODO: Delete test? no site-specific search?
+		it( 'should track theme query request fetching', () => {
+			const state = queryRequests( deepFreeze( {} ), {
+				type: THEMES_REQUEST,
+				siteId: 2916284,
+				query: { search: 'Hello' }
+			} );
+
+			expect( state ).to.eql( {
+				'2916284:{"search":"Hello"}': true
+			} );
+		} );
+
+		it( 'should track theme queries without specified site', () => {
+			const state = queryRequests( deepFreeze( {} ), {
+				type: THEMES_REQUEST,
+				query: { search: 'Hello' }
+			} );
+
+			expect( state ).to.eql( {
+				'{"search":"Hello"}': true
+			} );
+		} );
+
+		it( 'should accumulate queries', () => {
+			const original = deepFreeze( {
+				'2916284:{"search":"Hello"}': true
+			} );
+
+			const state = queryRequests( original, {
+				type: THEMES_REQUEST,
+				siteId: 2916284,
+				query: { search: 'Hello W' }
+			} );
+
+			expect( state ).to.eql( {
+				'2916284:{"search":"Hello"}': true,
+				'2916284:{"search":"Hello W"}': true
+			} );
+		} );
+
+		it( 'should track theme query request success', () => {
+			const state = queryRequests( deepFreeze( {} ), {
+				type: THEMES_REQUEST_SUCCESS,
+				siteId: 2916284,
+				query: { search: 'Mood' },
+				found: 1,
+				themes: [
+					mood
+				]
+			} );
+
+			expect( state ).to.eql( {
+				'2916284:{"search":"Mood"}': false
+			} );
+		} );
+
+		it( 'should track theme query request failure', () => {
```

Comment:
Hmm. Well essentially, we're covering the `queryRequests` reducer's reaction to different actions here, and the only information it really stores is if a request is ongoing, or not. And both `THEMES_REQUEST_SUCCESS` and `THEMES_REQUEST_FAILURE` signal that the request has finished (as oppsed to ` THEMES_REQUEST`, which we check to set the flag to `true`), so I actually think it kinda makes sense. This is just a consequence of what reducer stores which piece of information when a given action is triggered -- which in case of `queryRequests` clearly isn't a lot 😄 

To put it differently, I wouldn't know how else to cover [this functionality](https://github.com/Automattic/wp-calypso/pull/9476/files#diff-40ab1217977ee8f85be34d691f55b624R77) with tests.